File: app/src/main/java/ru/michanic/mymot/Protocols/ApiInterface.java
package ru.michanic.mymot.Protocols;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.POST;
import retrofit2.http.Query;
import ru.michanic.mymot.Models.AppPageText;
import ru.michanic.mymot.Models.Category;
import ru.michanic.mymot.Models.JsonResult;
import ru.michanic.mymot.Models.Location;
import ru.michanic.mymot.Models.Manufacturer;
import ru.michanic.mymot.Models.ModelDetails;
import ru.michanic.mymot.Models.Volume;

public interface ApiInterface {

    @POST("config.php?type=word_exceptions")
    Call<List<String>> loadExteptedWords();

    @POST("config.php?type=about")
    Call<AppPageText> loadAboutText();

    @POST("config.php?type=agreement")
    Call<AppPageText> loadAgreementText();


    @POST("catalog.php?type=only_regions")
    Call<List<Location>> loadRegions();

    @POST("catalog.php?type=cities")
    Call<List<Location>> loadRegionCities(@Query("id") int regionId);

    @POST("catalog.php?type=volumes")
    Call<List<Volume>> loadVolumes();

    @POST("catalog.php?type=classes")
    Call<List<Category>> loadClasses();

    @POST("catalog.php?type=models")
    Call<List<Manufacturer>> loadModels();

    @POST("catalog.php?type=model_details")
    Call<ModelDetails> loadModelDetails(@Query("id") int modelId);

}

File: app/src/main/java/ru/michanic/mymot/Kotlin/Protocols/ApiInterface.kt
package ru.michanic.mymot.Kotlin.Protocols

import retrofit2.Call
import retrofit2.http.POST
import retrofit2.http.Query
import ru.michanic.mymot.Kotlin.Models.*

interface ApiInterface {
    @POST("config.php?type=word_exceptions")
    fun loadExteptedWords(): Call<List<String?>?>?

    @POST("config.php?type=about")
    fun loadAboutText(): Call<AppPageText?>?

    @POST("config.php?type=agreement")
    fun loadAgreementText(): Call<AppPageText?>?

    @POST("catalog1.php?type=only_regions")
    fun loadRegions(): Call<List<Location?>?>?

    @POST("catalog1.php?type=cities")
    fun loadRegionCities(@Query("id") regionId: Int): Call<List<Location?>?>?

    @POST("catalog1.php?type=volumes")
    fun loadVolumes(): Call<List<Volume?>?>?

    @POST("catalog1.php?type=classes")
    fun loadClasses(): Call<List<Category?>?>?

    @POST("catalog1.php?type=models")
    fun loadModels(): Call<List<Manufacturer?>?>?

    @POST("catalog1.php?type=model_details")
    fun loadModelDetails(@Query("id") modelId: Int): Call<ModelDetails?>?

    @POST("catalog1.php?type=property_enums")
    fun loadPropertyEnums(@Query("id") modelId: Int): Call<Model>
}